framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceJavaParserStorage.java

```diff
@@ -762,18 +762,23 @@ private void addClass(
               className = classNameKey;
             } else {
               TypeElement classElt = TreeUtils.elementFromDeclaration(tree);
-              className = ElementUtils.getBinaryName(classElt);
-
-              for (TypeElement supertypeElement : ElementUtils.getSuperTypes(classElt, elements)) {
-                String supertypeName = ElementUtils.getBinaryName(supertypeElement);
-                @SuppressWarnings({"signature:assignment", "signature:return"}) // #979?
-                Set<String> supertypeSet =
-                    supertypesMap.computeIfAbsent(className, k -> new TreeSet<>());
-                supertypeSet.add(supertypeName);
-                @SuppressWarnings({"signature:assignment", "signature:return"}) // #979?
-                Set<String> subtypeSet =
-                    subtypesMap.computeIfAbsent(supertypeName, k -> new TreeSet<>());
-                subtypeSet.add(className);
+              if (classElt == null) {
```

Comment:
This change violates the method's documentation, which states that an element for `tree` (i.e., `classElt`) always exists (and therefore is non-null) if `classNameKey` is null.

Looking at the method's call sites, it looks like the expectation (e.g., at around line 714 above in `processNewClass`) is that in the cases where it's possible that there's no element, it should be handled by the caller.

I agree that this contract isn't great, but I don't think this change is the right one: it breaks the invariant that this method is clearly trying to preserve.

I think we should either:
* change the contract of the method to remove the relationship between classNameKey and the element. If we do this, we should probably remove `classNameKey` as a parameter all together, and move the code in `processNewClass` (and any other call sites) into this method.
* figure out which call site is breaking the invariant, and make an associated change there to fix this crash rather than here

Reply:
Thanks for the feedback. This null originated from the other `addClass` method below this one. To address this issue, I have moved the class name computation into this method when `classElt` is null..